src/xbot/common/controls/sensors/MockGamepad.java

```diff
@@ -1,40 +1,158 @@
 
-    MockJoystick mockLeftStick = new MockJoystick();
-    MockJoystick mockRightStick = new MockJoystick();
-    MockJoystick mockDpad = new MockJoystick();
-    MockJoystick mockLeftTrigger = new MockJoystick();
-    MockJoystick mockRightTrigger = new MockJoystick();
+    XYPair leftJoystickAxis;
+    XYPair rightJoystickAxis;
+    double leftTriggerAxis;
+    double rightTriggerAxis;
+    boolean leftBumper;
+    boolean rightBumper;
+    boolean AButton;
+    boolean BButton;
+    boolean XButton;
+    boolean YButton;
+    boolean backButton;
+    boolean startButton;
+    boolean leftStickButton;
+    boolean rightStickButton;
     
-    public MockGamepad() {
-        
+    public void setLeftStick(double x, double y){
+        leftJoystickAxis = new XYPair(x, y);
     }
     
-    @Override
-    public XJoystick getLeftStick() {
-        return mockLeftStick;
+    public void setRightStick(double x, double y){
+        rightJoystickAxis = new XYPair(x, y);
+    }
+    
+    public void setLeftTriggerValue(double x){
+        leftTriggerAxis = x;
+    }
+    
+    public void setRightTriggerValue(double x){
+        rightTriggerAxis = x;
     }
 
-    @Override
-    public XJoystick getRightStick() {
-        return mockRightStick;
+    public void setLeftBumper(boolean pressed){
+        leftBumper = pressed;
     }
     
-    @Override
-    public XJoystick getDpad() {
-        return mockDpad;
+    public void setRightBumper(boolean pressed){
+        rightBumper = pressed;
+    }
+    
+    public void setAButton(boolean pressed){
+        AButton = pressed;
+    }
+    
+    public void setBButton(boolean pressed){
+        BButton = pressed;
+    }
+    
+    public void setXButton(boolean pressed){
+        XButton = pressed;
+    }
+    
+    public void setYButton(boolean pressed){
+        YButton = pressed;
+    }
+    
+    public void setBackButton(boolean pressed){
+        backButton = pressed;
+    }
+    
+    public void setStartButton(boolean pressed){
+        startButton = pressed;
+    }
+    
+    public void setLeftStickButton(boolean pressed){
+        leftStickButton = pressed;
+    }
+    
+    public void setRightStickButton(boolean pressed){
+        rightStickButton = pressed;
+    }
+    
+    public XYPair getLeftStick(){
+        return leftJoystickAxis;
+    }
+    
+    public XYPair getRightStick(){
+        return rightJoystickAxis;
+    }
+    
+  
+    public double getLeftStickX() {
+        return leftJoystickAxis.x;
+    }
+    
+  
+    public double getRightStickX() {
+        return rightJoystickAxis.x;
     }
 
-    @Override
-    public XJoystick getLeftTrigger() {
-        return mockLeftTrigger;
+  
+    public double getLeftStickY() {
+        return leftJoystickAxis.y;
     }
 
-    @Override
-    public XJoystick getRightTrigger() {
-        return mockRightTrigger;
+  
+    public double getRightStickY() {
+        return rightJoystickAxis.y;
```

Comment:
Thought experiment: what happens if you call `getRightStickY` without first calling `setRightStick`? Hint: 💥